include/seqan3/io/sam_file/format_bam.hpp

```diff
@@ -775,38 +684,20 @@ inline void format_bam::write_alignment_record([[maybe_unused]] stream_type & st
         // ---------------------------------------------------------------------
         int32_t ref_length{};
 
-        // if alignment is non-empty, replace cigar_vector.
-        // else, compute the ref_length from given cigar_vector which is needed to fill field `bin`.
+        // Compute the ref_length from given cigar_vector which is needed to fill field `bin`.
         if (!std::ranges::empty(cigar_vector))
         {
             int32_t dummy_seq_length{};
             for (auto & [count, operation] : cigar_vector)
                 detail::update_alignment_lengths(ref_length, dummy_seq_length, operation.to_char(), count);
         }
-        else if (!std::ranges::empty(get<0>(align)) && !std::ranges::empty(get<1>(align)))
-        {
-            ref_length = std::ranges::distance(get<1>(align));
-
-            // compute possible distance from alignment end to sequence end
-            // which indicates soft clipping at the end.
-            // This should be replaced by a free count_gaps function for
-            // aligned sequences which is more efficient if possible.
-            int32_t off_end{static_cast<int32_t>(std::ranges::distance(seq)) - offset};
-
-            for (auto chr : get<1>(align))
-                if (chr == gap{})
-                    ++off_end;
-
-            off_end -= ref_length;
-            cigar_vector = detail::get_cigar_vector(align, offset, off_end);
-        }
 
         if (cigar_vector.size() >= (1 << 16)) // must be written into the sam tag CG
         {
             tag_dict["CG"_tag] = detail::get_cigar_string(cigar_vector);
             cigar_vector.resize(2);
             cigar_vector[0] = cigar{static_cast<uint32_t>(std::ranges::distance(seq)), 'S'_cigar_operation};
-            cigar_vector[1] = cigar{static_cast<uint32_t>(std::ranges::distance(get<1>(align))), 'N'_cigar_operation};
+            cigar_vector[1] = cigar{static_cast<uint32_t>(ref_length), 'N'_cigar_operation};
```

Comment:
We don't really check the ref or query length in general, no?
`update_alignment_lengths` doesn't check for overflow.
But this is a general problem and part of this PR.

Reply:
no we don't so far :/ chances are low though that you have a that long alignment that you want to write to SAM/BAM. Those are usually used for read alignments not genome alignments